tdrs-backend/tdpservice/parsers/validators.py

```diff
@@ -61,16 +127,124 @@ def startsWith(substring):
+    """Validate that value is alphanumeric."""
+    return make_validator(
+        lambda value: value.isalnum(),
+        lambda value: f'{value} is not alphanumeric.'
+    )
+
+def isBlank():
+    """Validate that string value is blank."""
+    return make_validator(
+        lambda value: value.isspace(),
+        lambda value: f'{value} is not blank.'
+    )
+
 def notEmpty(start=0, end=None):
     """Validate that string value isn't only blanks."""
     return make_validator(
-        lambda value: not value[start:end if end else len(value)].isspace(),
-        lambda value: f'{value} contains blanks between positions {start} and {end if end else len(value)}.'
+        lambda value: not str(value)[start:end if end else len(str(value))].isspace(),
+        lambda value: f'{str(value)} contains blanks between positions {start} and {end if end else len(str(value))}.'
+    )
+
+def isEmpty(start=0, end=None):
+    """Validate that string value is only blanks."""
+    return make_validator(
+        lambda value: value[start:end if end else len(value)].isspace(),
+        lambda value: f'{value} is not blank between positions {start} and {end if end else len(value)}.'
+    )
+
+
+def notZero(number_of_zeros=1):
+    """Validate that value is not zero."""
+    return make_validator(
+        lambda value: value != '0' * number_of_zeros,
+        lambda value: f'{value} is zero.'
+    )
+
+
+def isLargerThan(LowerBound):
+    """Validate that value is larger than the given value."""
+    return make_validator(
+        lambda value: float(value) > LowerBound if value is not None else False,
+        lambda value: f'{value} is not larger than {LowerBound}.'
+    )
+
+
+def isSmallerThan(UpperBound):
+    """Validate that value is smaller than the given value."""
+    return make_validator(
+        lambda value: value < UpperBound,
+        lambda value: f'{value} is not smaller than {UpperBound}.'
+    )
+
+def isLargerThanOrEqualTo(LowerBound):
+    """Validate that value is larger than the given value."""
+    return make_validator(
+        lambda value: value >= LowerBound,
+        lambda value: f'{value} is not larger than {LowerBound}.'
+    )
+
+def isSmallerThanOrEqualTo(UpperBound):
+    """Validate that value is smaller than the given value."""
+    return make_validator(
+        lambda value: value <= UpperBound,
+        lambda value: f'{value} is not smaller than {UpperBound}.'
+    )
+
+def isInLimits(LowerBound, UpperBound):
+    """Validate that value is in a range including the limits."""
+    return make_validator(
+        lambda value: value >= LowerBound and value <= UpperBound,
+        lambda value: f'{value} is not larger and equal to {LowerBound} and smaller and equal to {UpperBound}.'
     )
 
 
 # custom validators
 
+def month_year_monthIsValid():
+    """Validate that in a monthyear combination, the month is a valid month."""
+    return make_validator(
+        lambda value: int(str(value)[4:]) in range(1, 13),
```

Comment:
i believe `DATE_OF_BIRTH` is 8 characters, so this is throwing a false error on the month saying `0127` from `20191027` (in my case) is invalid. The month is `01` and the day is `27`

Reply:
I added a test and corrected the validator to only look for 2 digits of the month after year :)